app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt

```diff
@@ -237,6 +239,12 @@ class HomeFragment : Fragment() {
             )
         }
 
+        val applicationContext = requireContext().applicationContext
+        lifecycleScope.launch(Dispatchers.IO) {
```

Comment:
This is more of a curiosity, but what would be the difference here if we used `viewlifecycleScope` instead of `lifecycleScope`?

Reply:
I see `viewLifecycleScope` as rather a naming idiom we tend to use for parameters referencing `lifecycleScope`.

`lifecycleScope` exists for every `LifecycleOwner` - mostly Fragments and Activities implementations and exists beween the created and destroyed states.
We also use `view.toScope` as maybe an analogue to the above but for plain Views with the scope existing between View's attached / detached states.

Reply:
TL;DR: `viewLifecycleScope` exists during the lifetime of the views (onCreateView/OnDestroyView), while `lifecycleScope` exists during the lifetime of the whole Fragment. This gets only interesting when a Fragment gets detached (views get destroyed) and later reattached (new views get created). I think the rule of thumb is that when you schedule anything that needs views, then use `viewLifecycleScope` to not keep running if the views are no longer actually on the screen.

We use `View.toScope()` (which is similar to `viewLifecycleScope`) in A-C for two reasons:
* In features we have access to `View` objects that get passed in as parameters, but we never see the `Fragment` instance. FWIW we could be running in an Activity without a Fragment.
* We saw a very scary bug where `viewLifecycleScope` actually didn't get cancelled. 😱